lib/src/migrator.dart

```diff
@@ -0,0 +1,296 @@
+    super.visitImportRule(importRule);
+    if (importRule.imports.length != 1) {
+      throw Exception("Multiple imports in single rule not supported yet");
+    }
+    if (importRule.imports.first is! DynamicImport) return;
+    var import = importRule.imports.first as DynamicImport;
+
+    var potentialOverrides = <VariableDeclaration>[];
+    var isTopLevel = false;
+    for (var statement in currentSheet.children) {
+      if (statement == importRule) {
+        isTopLevel = true;
+        break;
+      }
+      if (statement is VariableDeclaration) {
+        potentialOverrides.add(statement);
+      }
+    }
+    if (!isTopLevel) {
+      // TODO(jathak): Handle nested imports
+      return;
+    }
+    var path = resolveImport(import.url, from: currentPath);
+    var importedSheetApi = _apis[path];
+    namespaces[findNamespace(import.url)] = path;
+
+    var overrides = potentialOverrides.where((declaration) =>
+        importedSheetApi.variables.containsKey(declaration.name) &&
+        importedSheetApi.variables[declaration.name].isGuarded);
+    var config = overrides
+        .map((decl) => "\$${decl.name}: ${decl.expression}")
+        .join(",\n  ");
+    if (config != "") config = " with (\n  $config\n)";
+    patches.add(Patch(importRule.span, '@use ${import.span.text}$config'));
+
+    if (_migrateDependencies) _migrationQueue.add(path);
+  }
+
+  /// Adds a namespace to a function call if it is necessary.
+  @override
+  void visitFunctionExpression(FunctionExpression node) {
+    super.visitFunctionExpression(node);
+    if (node.name.contents.length != 1 || node.name.contents.first is! String) {
+      // This is a plain CSS invocation if it is interpolated, so there's no
+      // need to namespace.
+      return;
+    }
+    var name = node.name.asPlain;
+    var ns = findNamespaceFor(name, ApiType.functions);
+    if (ns == null) {
+      ns = makeImplicitDependencyExplicit(name, ApiType.functions);
+      if (ns == null) {
+        if (_apis[currentPath].functions.containsKey(name)) return;
+        if (builtInFunctionModules.containsKey(name)) {
+          ns = Namespace(builtInFunctionModules[name]);
+          if (builtInFunctionNameChanges.containsKey(name)) {
+            name = builtInFunctionNameChanges[name];
+          }
+          if (!builtInModules.contains(ns.namespace)) {
+            builtInModules.add(ns.namespace);
+            patches.add(Patch.prepend('@use "sass:$ns";\n'));
+          }
+        }
+      }
+    }
+    patches.add(Patch(node.name.span, "$ns.$name"));
+  }
+
+  /// Adds a namespace to an include rule if it is necessary.
+  @override
+  void visitIncludeRule(IncludeRule node) {
+    var ns = findNamespaceFor(node.name, ApiType.mixins);
+    if (ns == null) {
+      ns = makeImplicitDependencyExplicit(node.name, ApiType.mixins);
+      if (ns == null) return;
+    }
+    var endName = node.arguments.span.start.offset;
+    var startName = endName - node.name.length;
```

Comment:
`node.span.start.offset` would start at `@include` instead of at the beginning of the name. I wanted to start at the name instead to only patch the name and allow the shorthand syntax for `.sass` to be preserved if it was there in the original stylesheet.